### PR TITLE
Fixes #12672 - Update katello url helper to be ruby 2.2.2 compat

### PR DESCRIPTION
**Pull request by cpeters**
The URI parsing behavior of ruby 2.2.2 has changed from 2.0 and 2.1

A url without a host is considered invalid.
When parsing a url without a host in 2.0 and 2.1 an exception is
thrown, caught, and false is returned. This is the expected
behavior.

When parsing a url without a host in 2.2.2 no exception is thrown.
Updating the method to explicitly check for the presense of a host;
restoring the intended behavior.

### CODE REVIEW

File: app/helpers/katello/katello_url_helper.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@ def file_prefix?(url)
     private
 
     def valid_for_prefixes(url, prefixes)
+      return false unless URI.parse(url).host.present?
       return false unless (scheme = URI.parse(url).scheme).present?
```
Comment (johnpmitsch) [inline app/helpers/katello/katello_url_helper.rb]: What about this?

```
return false if URI.parse(url).host.blank? || (scheme = URI.parse(url).scheme).blank?
```

Comment (cpeters) [inline app/helpers/katello/katello_url_helper.rb]: @johnpmitsch 

I'm not sure I agree with moving these to the same line. They are checking different parts of the URI. Also there is a nested assignment. This gets messy and hard to parse when reading back over the code, imo.

Comment (ehelms) [inline app/helpers/katello/katello_url_helper.rb]: I like separate conditions here due to the returns and not a general if
block.

File: app/helpers/katello/katello_url_helper.rb [hunk 2]
```diff
@@ -8,17 +8,24 @@ module KatelloUrlHelper
     end
 
     def kurl_valid?(url)
-      valid_for_prefixes(url, PROTOCOLS)
+      validate_host(url)
```
Comment (johnpmitsch) [inline app/helpers/katello/katello_url_helper.rb]: if validate_host(url) returns false, but validate_scheme(url, PROTOCOLS) returns true, this method will still return true because is returning the last line, right?

File: app/helpers/katello/katello_url_helper.rb [hunk 3]
```diff
@@ -8,17 +8,24 @@ module KatelloUrlHelper
...
-      valid_for_prefixes(url, [FILEPREFIX])
+      validate_scheme(url, [FILEPREFIX])
     end
 
     private
 
-    def valid_for_prefixes(url, prefixes)
-      return false unless (scheme = URI.parse(url).scheme).present?
+    def validate_host(url)
+      return false if URI.parse(url).host.blank?
```
Comment (ehelms) [inline app/helpers/katello/katello_url_helper.rb]: This can be shortened to:

```
def validate_host(url)
  URI.parse(url).host.present?
rescue URI::InvalidURIError
  false
end
```

File: app/helpers/katello/katello_url_helper.rb (hunk 2 above)
Comment (cpeters) [inline app/helpers/katello/katello_url_helper.rb]: Yes, good catch.!

**Comment by ehelms** [comment]
[test]

**Comment by cpeters** [comment]
Closing in favor of this; https://github.com/johnpmitsch/katello/pull/12
